docfx_yaml/extension.py

```diff
@@ -1260,6 +1261,117 @@ def find_markdown_pages(app, outdir):
             })
 
 
+# Finds and replaces occurrences which should be a cross reference in the given
+# content, except for the current name.
+def convert_cross_references(content, current_name, sorted_map):
```

Comment:
Is there a more specific name for `sorted_map`? From the name, I don't know what it holds. Also, why does it need to be sorted? And `map` seems like not the right name for a list.

Reply:
Ooooooops. Should've looked that more thoroughly :( Updated to reflect that it's a list.

I'm putting it into a sorted list so that the shorter version of the match does not take the cross reference conversion before the appropriate match. For example, without the sorted list in the following content:
```
var_type: google.cloud.bigquery_storage_v1.types.ReadSession
```
we expect the following:
```
var_type: <xref uid="google.cloud.bigquery_storage_v1.types.ReadSession">google.cloud.bigquery_storage_v1.types.ReadSession</xref>
```
however, we could and will encounter the following:
```
var_type: <xref uid="google.cloud.bigquery_storage_v1.types">google.cloud.bigquery_storage_v1.types</xref>.ReadSession
```

Looking back, this doesn't sound intuitive to me either but since we're looking for `keyword in word` and can't do an exact match and also know if there is more we should consider than the match we have (for example, when google.cloud.bigquery_storage_v1.types is a match how do we know there's more?), this is the best I can think of for now. Happy to explore better alternatives!